src/freertos_drivers/common/TMAG5273.hxx

```diff
@@ -0,0 +1,555 @@
+        DIAG_STATUS_MASK   = 0x02,
+        /// Conversion, 0 = conversion not complete, 1 = conversion complete
+        RESULT_STATUS_MASK = 0x01,
+    };
+
+    /// Conversion oversampling average configuration
+    enum class ConversionAverage : uint8_t
+    {
+        AVG_1  = (0 << 2), ///< average 1 saple
+        AVG_2  = (1 << 2), ///< average 2 saples
+        AVG_4  = (2 << 2), ///< average 4 saples
+        AVG_8  = (3 << 2), ///< average 8 saples
+        AVG_16 = (4 << 2), ///< average 16 saples
+        AVG_32 = (5 << 2), ///< average 32 saples
+    };
+
+    /// Angle calculation to enable.
+    enum class AngleEnable : uint8_t
+    {
+        OFF        = (0 << 2), ///< angle not enabled
+        XY_ENABLE  = (1 << 2), ///< X + Y derived angle
+        YZ_ENABLE  = (2 << 2), ///< Y + Z derived angle
+        XZ_ENABLE  = (3 << 2), ///< X + Z derived angle
+    };
+
+    /// Operating modes of the device.
+    enum class OperatingMode : uint8_t
+    {
+        STANDBY           = 0x00, ///< starts new conversion at trigger event
+        SLEEP             = 0x01, ///< sleep
+        CONTINUOUS        = 0x02, ///< continuous measurement mode
+        WAKE_UP_AND_SLEEP = 0x03, ///< wakeu-up and slee mode
+    };
+
+    /// Constructor.
+    /// @param address is the 7-bit address (right aligned), typically 0x35,
+    ///        0x22, 0x78 or 0x44). Default is the A1/A2 device.
+    TMAG5273(uint8_t address = 0x35)
+        : i2cAddress_(address)
+    { }
+
+    /// Constructor. Can only be called from thread context.
+    /// @param i2c_path path to the i2c bus to use.
+    /// @param address is the 7-bit address (right aligned), typically 0x35,
+    ///        0x22, 0x78 or 0x44). Default is the A1/A2 device.
+    TMAG5273(const char *i2c_path, uint8_t address = 0x35)
+        : i2cAddress_(address)
+    {
+        init(i2c_path);
+    }
+
+    /// Destructor.
+    ~TMAG5273()
+    {
+        if (fd_ >= 0)
+        {
+            ::close(fd_);
+        }
+    }
+
+    void init(const char *i2c_path)
+    {
+        int fd = ::open(i2c_path, O_RDWR);
+        HASSERT(fd >= 0);
+        init(fd);
+    }
+
+    /// Initializes the device.
+    /// @param i2c_fd is the file descriptor of the i2c port to use.
+    void init(int i2c_fd)
+    {
+        fd_ = i2c_fd;
+    }
+
+    friend class MagSensorTest;
+    friend class Input;
+
+    /// Checks whether the magnetic sensor is present.
+    /// @return device identifier, else ID_UNKNOWN if not detected or uknown
+    ///         device
+    DeviceID is_present()
+    {
+        uint8_t rc[3];
+        // We start with a dummy read to ensure I2C is in a known state.
+        register_read(TMAG5273::DEVICE_ID, rc, 3);
+        if (register_read(TMAG5273::DEVICE_ID, rc, 3) < 0)
+        {
+            // Error reading from i2c.
+            LOG(VERBOSE, "error read");
+            return DeviceID::ERROR;
+        }
+        if ((rc[0] != static_cast<uint8_t>(DeviceID::MT_40_AND_80) &&
```

Comment:
Need to mask off the reserved bits 7-2 of the Device ID register. These reserved bits may be set, confirmed by TI.

Reply:
Fixed.